crates/matrix-sdk-sqlite/src/event_cache_store.rs

```diff
@@ -486,24 +486,27 @@ impl EventCacheStore for SqliteEventCacheStore {
                     }
 
                     Update::PushItems { at, items } => {
+                        if items.is_empty() {
+                            // Should never happens, but better be safe.
+                            continue;
+                        }
+
                         let chunk_id = at.chunk_identifier().index();
 
                         trace!(%room_id, "pushing {} items @ {chunk_id}", items.len());
 
-                        for (i, event) in items.into_iter().enumerate() {
+                        let mut statement = txn.prepare(
+                            "INSERT INTO events(chunk_id, room_id, event_id, content, position) VALUES (?, ?, ?, ?, ?)"
+                        )?;
+
+                        for (nth, event) in items.into_iter().enumerate() {
```

Comment:
Please keep the previous name `i` for `nth`. It's the simplest index variable, there's absolutely no reason to rename it, everyone expects "i" to be a loop index, "nth" is surprising and unsettling.